source/solvers/tracer.cc

```diff
@@ -712,6 +712,44 @@ Tracer<dim>::setup_dofs()
                                        &this->previous_solutions);
 }
 
+template <int dim>
+void
+Tracer<dim>::update_boundary_conditions()
+{
+  if (!this->simulation_parameters.boundary_conditions_tracer.time_dependent)
+    return;
+
+  double time = this->simulation_control->get_current_time();
+  {
+    nonzero_constraints.clear();
+    DoFTools::make_hanging_node_constraints(this->dof_handler,
+                                            nonzero_constraints);
+
+    for (unsigned int i_bc = 0;
+         i_bc < this->simulation_parameters.boundary_conditions_tracer.size;
+         ++i_bc)
+      {
+        // Dirichlet condition : imposed temperature at i_bc
+        if (this->simulation_parameters.boundary_conditions_tracer.type[i_bc] ==
+            BoundaryConditions::BoundaryType::tracer_dirichlet)
+          {
+            this->simulation_parameters.boundary_conditions_tracer.tracer[i_bc]
+              ->set_time(time);
+            VectorTools::interpolate_boundary_values(
+              this->dof_handler,
+              this->simulation_parameters.boundary_conditions_tracer.id[i_bc],
+              *this->simulation_parameters.boundary_conditions_tracer
+                 .tracer[i_bc],
+              nonzero_constraints);
+          }
+      }
+  }
```

Comment:
Why is there this additional scope here, it's not necessary no?

Reply:
I put it because that's the way that it was written in the setup_dofs functions, but without this additional scope it works just as well so I'll remove it.